### ISSUE TITLE
Qwen1.5 - TypeError: argument 'tokens': 'NoneType' object cannot be converted to 'PyString'

### ISSUE THREAD

**Issue by GennVa** [vllm-project/vllm#3118]
Hello,
I'm using Qwen1.5, and after a finetuning (with Lora), I have this error, using vllm:
`TypeError: argument 'tokens': 'NoneType' object cannot be converted to 'PyString'`

This is the main code:
```
LLM(model=path, tokenizer=path, tensor_parallel_size=1, gpu_memory_utilization=0.75)
sampling_params = SamplingParams(
            n=1,
            best_of=None,
            presence_penalty=0.0,
            frequency_penalty=0.0,
            temperature=0.7,
            top_p=1.0,
            top_k=-1,
            use_beam_search=False,
            length_penalty=1.0,
            early_stopping=False,
            stop="<EOSQ>",
            stop_token_ids=None,
            ignore_eos=False,
            max_tokens=2048,
            logprobs=None,
            skip_special_tokens=True
        )
```

I tried to get the lengths between tokenizer and vocab_size, and they differ by 5 tokens (151648 vs 151643).
I tried manually changing the tokenizer's vocab_size to be the same, but the error remains.
Thanks

Related to: #340 #2398

**Comment by 777ki** [comment]
do you have the full stack backtrace info?

**Comment by Chenghao-Jia** [comment]
I have the same error. This is the full stack backtrace info.
```
INFO 03-02 14:33:34 config.py:413] Custom all-reduce kernels are temporarily disabled due to stability issues. We will re-enable them once the issues are resolved.
2024-03-02 14:33:37,595	INFO worker.py:1724 -- Started a local Ray instance.
INFO 03-02 14:33:38 llm_engine.py:79] Initializing an LLM engine with config: model='/home/chenghao/workspace/Oral_calculation/model/1-grade/Qwen1.5-0.5B-Chat', tokenizer='/home/chenghao/workspace/Oral_calculation/model/1-grade/Qwen1.5-0.5B-Chat', tokenizer_mode=auto, revision=None, tokenizer_revision=None, trust_remote_code=True, dtype=torch.bfloat16, max_seq_len=32768, download_dir=None, load_format=auto, tensor_parallel_size=2, disable_custom_all_reduce=True, quantization=None, enforce_eager=True, kv_cache_dtype=auto, device_config=cuda, seed=2)
Special tokens have been added in the vocabulary, make sure the associated word embeddings are fine-tuned or trained.
Special tokens have been added in the vocabulary, make sure the associated word embeddings are fine-tuned or trained.
INFO 03-02 14:33:49 llm_engine.py:337] # GPU blocks: 25975, # CPU blocks: 5461
Processed prompts:   0%|                                                               | 0/28903 [00:00<?, ?it/s]Traceback (most recent call last):
  File "/sshfs/chenghao/Oral_calculation/vllm/predict.py", line 31, in <module>
    outputs = llm.generate(prompts, sampling_params)
  File "/home/chenghao/yes/envs/vllm/lib/python3.10/site-packages/vllm/entrypoints/llm.py", line 182, in generate
    return self._run_engine(use_tqdm)
  File "/home/chenghao/yes/envs/vllm/lib/python3.10/site-packages/vllm/entrypoints/llm.py", line 208, in _run_engine
    step_outputs = self.llm_engine.step()
  File "/home/chenghao/yes/envs/vllm/lib/python3.10/site-packages/vllm/engine/llm_engine.py", line 832, in step
    return self._process_model_outputs(output, scheduler_outputs)
  File "/home/chenghao/yes/envs/vllm/lib/python3.10/site-packages/vllm/engine/llm_engine.py", line 735, in _process_model_outputs
    self._process_sequence_group_outputs(seq_group, outputs)
  File "/home/chenghao/yes/envs/vllm/lib/python3.10/site-packages/vllm/engine/llm_engine.py", line 605, in _process_sequence_group_outputs
    self._decode_sequence(seq, seq_group.sampling_params)
  File "/home/chenghao/yes/envs/vllm/lib/python3.10/site-packages/vllm/engine/llm_engine.py", line 908, in _decode_sequence
    read_offset) = detokenize_incrementally(
  File "/home/chenghao/yes/envs/vllm/lib/python3.10/site-packages/vllm/transformers_utils/tokenizer.py", line 221, in detokenize_incrementally
    new_text = tokenizer.convert_tokens_to_string(
  File "/home/chenghao/yes/envs/vllm/lib/python3.10/site-packages/transformers/tokenization_utils_fast.py", line 612, in convert_tokens_to_string
    return self.backend_tokenizer.decoder.decode(tokens)
TypeError: argument 'tokens': 'NoneType' object cannot be converted to 'PyString'
```

**Comment by whi497** [comment]
same error
```
Exception in callback functools.partial(<function _raise_exception_on_finish at 0x7f7e93abc790>, request_tracker=<vllm.engine.async_llm_engine.RequestTracker object at 0x7f7e8f4f1b80>)
handle: <Handle functools.partial(<function _raise_exception_on_finish at 0x7f7e93abc790>, request_tracker=<vllm.engine.async_llm_engine.RequestTracker object at 0x7f7e8f4f1b80>)>
Traceback (most recent call last):
  File "/root/miniconda3/envs/myenv/lib/python3.9/site-packages/vllm/engine/async_llm_engine.py", line 29, in _raise_exception_on_finish
    task.result()
  File "/root/miniconda3/envs/myenv/lib/python3.9/site-packages/vllm/engine/async_llm_engine.py", line 414, in run_engine_loop
    has_requests_in_progress = await self.engine_step()
  File "/root/miniconda3/envs/myenv/lib/python3.9/site-packages/vllm/engine/async_llm_engine.py", line 393, in engine_step
    request_outputs = await self.engine.step_async()
  File "/root/miniconda3/envs/myenv/lib/python3.9/site-packages/vllm/engine/async_llm_engine.py", line 203, in step_async
    return self._process_model_outputs(output, scheduler_outputs)
  File "/root/miniconda3/envs/myenv/lib/python3.9/site-packages/vllm/engine/llm_engine.py", line 756, in _process_model_outputs
    self._process_sequence_group_outputs(seq_group, outputs)
  File "/root/miniconda3/envs/myenv/lib/python3.9/site-packages/vllm/engine/llm_engine.py", line 626, in _process_sequence_group_outputs
    self._decode_sequence(seq, seq_group.sampling_params)
  File "/root/miniconda3/envs/myenv/lib/python3.9/site-packages/vllm/engine/llm_engine.py", line 932, in _decode_sequence
    read_offset) = detokenize_incrementally(
  File "/root/miniconda3/envs/myenv/lib/python3.9/site-packages/vllm/transformers_utils/tokenizer.py", line 221, in detokenize_incrementally
    new_text = tokenizer.convert_tokens_to_string(
  File "/root/miniconda3/envs/myenv/lib/python3.9/site-packages/transformers/tokenization_utils_fast.py", line 612, in convert_tokens_to_string
    return self.backend_tokenizer.decoder.decode(tokens)
TypeError: argument 'tokens': 'NoneType' object cannot be converted to 'PyString'
```

**Comment by GennVa** [comment]
@777ki I confirm these backtraces in my case too

**Comment by darrenangle** [comment]
same...

**Comment by Xls1994** [comment]
same error when I use the api client

`INFO:     127.0.0.1:43128 - "POST /v1/chat/completions HTTP/1.1" 500 Internal Server Error
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "/opt/conda/lib/python3.10/site-packages/vllm/engine/async_llm_engine.py", line 29, in _raise_exception_on_finish
    task.result()
  File "/opt/conda/lib/python3.10/site-packages/vllm/engine/async_llm_engine.py", line 414, in run_engine_loop
    has_requests_in_progress = await self.engine_step()
  File "/opt/conda/lib/python3.10/site-packages/vllm/engine/async_llm_engine.py", line 393, in engine_step
    request_outputs = await self.engine.step_async()
  File "/opt/conda/lib/python3.10/site-packages/vllm/engine/async_llm_engine.py", line 203, in step_async
    return self._process_model_outputs(output, scheduler_outputs)
  File "/opt/conda/lib/python3.10/site-packages/vllm/engine/llm_engine.py", line 756, in _process_model_outputs
    self._process_sequence_group_outputs(seq_group, outputs)
  File "/opt/conda/lib/python3.10/site-packages/vllm/engine/llm_engine.py", line 626, in _process_sequence_group_outputs
    self._decode_sequence(seq, seq_group.sampling_params)
  File "/opt/conda/lib/python3.10/site-packages/vllm/engine/llm_engine.py", line 932, in _decode_sequence
    read_offset) = detokenize_incrementally(
  File "/opt/conda/lib/python3.10/site-packages/vllm/transformers_utils/tokenizer.py", line 221, in detokenize_incrementally
    new_text = tokenizer.convert_tokens_to_string(
  File "/opt/conda/lib/python3.10/site-packages/transformers/tokenization_utils_fast.py", line 612, in convert_tokens_to_string
    return self.backend_tokenizer.decoder.decode(tokens)
TypeError: argument 'tokens': 'NoneType' object cannot be converted to 'PyString'`

**Comment by youkaichao** [comment]
Hi, friends, can you please try to build from source with the latest main? I think https://github.com/vllm-project/vllm/pull/3685 should fix this problem.

**Comment by GennVa** [comment]
> Hi, friends, can you please try to build from source with the latest main? I think #3685 should fix this problem.

Solved with v0.4.0.post1

**Comment by Greatpanc** [comment]
File "/mnt/bn/pan-personal-bytenas-yg/yinchangpan/software/conda/envs/ycp_py311_mfip/lib/python3.11/site-packages/lmdeploy/tokenizer.py", line 569, in detokenize_incrementally
    return self.model.detokenize_incrementally(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/mnt/bn/pan-personal-bytenas-yg/yinchangpan/software/conda/envs/ycp_py311_mfip/lib/python3.11/site-packages/lmdeploy/tokenizer.py", line 448, in detokenize_incrementally
    new_text = self._convert_tokens_to_string_with_added_encoders(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/mnt/bn/pan-personal-bytenas-yg/yinchangpan/software/conda/envs/ycp_py311_mfip/lib/python3.11/site-packages/lmdeploy/tokenizer.py", line 373, in _convert_tokens_to_string_with_added_encoders
    return tokenizer.convert_tokens_to_string(output_tokens)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/mnt/bn/pan-personal-bytenas-yg/yinchangpan/software/conda/envs/ycp_py311_mfip/lib/python3.11/site-packages/transformers/tokenization_utils_fast.py", line 620, in convert_tokens_to_string
    
TypeError: argument 'tokens': 'NoneType' object cannot be converted to 'PyString'
[TM][INFO] ~LlamaBatch()
[TM][INFO] [InternalThreadEntry] stop requested.
[TM][INFO] [OutputThreadEntry] stop requested. 
I had the same problem on qwen2-1.5b model and my lmdeploy version is 0.4.2
